cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -45,12 +83,99 @@ class CoursewareSearchIndexer(object):
+    #      (True) or is a property directly on the course object
+    ABOUT_INFORMATION_TO_INCLUDE = [
+        ("advertised_start", PROPERTY, False),
+        ("announcement", PROPERTY, True),
+        ("start", PROPERTY, False),
+        ("end", PROPERTY, False),
+        ("effort", PROPERTY, True),
+        ("display_name", ANALYSE, False),
+        ("overview", ANALYSE, True),
+        ("title", ANALYSE | PROPERTY, True),
+        ("university", ANALYSE | PROPERTY, True),
+        ("number", ANALYSE | PROPERTY, False),
+        ("short_description", ANALYSE, True),
+        ("description", ANALYSE, True),
+        ("key_dates", ANALYSE, True),
+        ("video", ANALYSE, True),
+        ("course_staff_short", ANALYSE, True),
+        ("course_staff_extended", ANALYSE, True),
+        ("requirements", ANALYSE, True),
+        ("syllabus", ANALYSE, True),
+        ("textbook", ANALYSE, True),
+        ("faq", ANALYSE, True),
+        ("more_info", ANALYSE, True),
+        ("ocw_links", ANALYSE, True),
+        ("enrollment_start", PROPERTY, False),
+        ("enrollment_end", PROPERTY, False),
+    ]
+
+    @classmethod
+    def index_about_information(cls, modulestore, course):
+        """
+        Add the given course to the course discovery index
+
+        Arguments:
+        modulestore - modulestore object to use for operations
+
+        course - course object from which to take properties, locate about information
+        """
+        searcher = SearchEngine.get_search_engine(INDEX_NAME)
+        if 'DISABLE_START_DATES' not in settings.FEATURES:
+            settings.FEATURES['DISABLE_START_DATES'] = False
+        if not searcher:
+            return
+
+        course_id = unicode(course.id)
+        try:
+            course_info = {
+                'id': course_id,
+                'course': course_id,
+                'content': {},
+                'image_url': course_image_url(course),
+            }
+
+            for about_information in cls.ABOUT_INFORMATION_TO_INCLUDE:
+                section_content = fetch_course_information_detail(modulestore, course, about_information)
+                if section_content:
+                    if isinstance(section_content, basestring):
+                        section_content = strip_html_content_to_text(section_content)
+                    if about_information[1] & cls.ANALYSE:
+                        course_info['content'][about_information[0]] = section_content
+                    if about_information[1] & cls.PROPERTY:
+                        course_info[about_information[0]] = section_content
+
+            searcher.index(DISCOVERY_DOCUMENT_TYPE, course_info)
+        except Exception as err:
+            log.exception(
+                "Course discovery indexing error encountered, course discovery index may be out of date %s - %r",
+                course_id,
+                err
+            )
+            raise
+        else:
```

Comment:
Or, if you move the try to just around the search indexing work, you can leave it in the else